src/core/net/ip6.cpp

```diff
@@ -603,27 +622,240 @@ otError Ip6::HandleOptions(Message &aMessage, Header &aHeader, bool &aForward)
     return error;
 }
 
-otError Ip6::HandleFragment(Message &aMessage)
+otError Ip6::HandleFragmentation(Message &aMessage, IpProto aIpProto)
+{
+    otError error = OT_ERROR_NONE;
+
+#if OPENTHREAD_ENABLE_IP6_FRAGMENTATION
+    Header         header;
+    FragmentHeader fragmentHeader;
+    Message *      fragment           = NULL;
+    uint16_t       fragmentCnt        = 0;
+    uint16_t       payloadFragment    = 0;
+    uint16_t       offset             = 0;
+    uint32_t       maxPayloadFragment = (1028 - aMessage.GetOffset() - sizeof(fragmentHeader)) & 0xfffffff8;
```

Comment:
nit: avoid magic numbers

Reply:
have created a definition and a constant.